presto-main/src/main/java/com/facebook/presto/metadata/SignatureBinder.java

```diff
@@ -445,4 +399,229 @@ private static TypeSignatureParameter applyBoundVariables(TypeSignatureParameter
                 throw new IllegalStateException("Unknown parameter kind: " + parameter.getKind());
         }
     }
+
+    private static List<TypeSignature> expandVarargFormalTypeSignature(List<TypeSignature> formalTypeSignatures, int actualArity)
+    {
+        int variableArityArgumentsCount = actualArity - formalTypeSignatures.size();
+        if (variableArityArgumentsCount == -1) {
+            return formalTypeSignatures.subList(0, formalTypeSignatures.size() - 1);
+        }
+        if (variableArityArgumentsCount == 0) {
+            return formalTypeSignatures;
+        }
+        checkArgument(variableArityArgumentsCount > 0 && !formalTypeSignatures.isEmpty());
```

Comment:
Is this check appropriate? It looks like it's going to cause an `IllegalArgumentException` in `appendArgumentConstraintsTo` if you try to do something like `my_vararg(x, y, z...)` and call it like `my_vararg(1)`